Introduction to Passwords Knowledge Base Article (#1708)
privacyguides/privacyguides.org: docs/basics/passwords-overview.en.md

---
title: "Introduction to Passwords"
icon: 'material/shield-key-outline'
---

> Passwords are an essential part of our everyday digital lives. We use them to protect our accounts, our devices and our secrets. Despite often being the only thing between us and an adversary who's after our private information, not a lot of thought is put into them which often leads to people using passwords that can be easily guessed or brute-forced.

Reviewer: ```suggestion
Passwords are an essential part of our everyday digital lives. We use them to protect our accounts, our devices and our secrets. Despite often being the only thing between us and an adversary who's after our private information, not a lot of thought is put into them, which often leads to people using passwords that can be easily guessed or brute-forced.
```

## Best Practices

### Use unique passwords for every service

Imagine this; you sign up for an account with the same e-mail and password on multiple online services. If one of those service providers is malicious, or their service has a data breach that exposes your password in an unencrypted format, all a bad actor would have to do is try that e-mail and password combination across multiple popular services until they get a hit. It doesn't matter how strong that one password is, because they already have it.

> This is called [credential stuffing](https://en.wikipedia.org/wiki/Credential_stuffing) and it is one of the most common ways that your accounts can be compromised by bad actors. To avoid this, make sure that you never re-use your passwords.

Reviewer: ```suggestion
This is called [credential stuffing](https://en.wikipedia.org/wiki/Credential_stuffing), and it is one of the most common ways that your accounts can be compromised by bad actors. To avoid this, make sure that you never re-use your passwords.
```

### Use randomly generated passwords

==You should **never** rely on yourself to come up with a good password.== We recommend using [randomly generated passwords](#passwords) or [diceware passphrases](#diceware) with sufficient entropy to protect your accounts and devices.

All of our [recommended password managers](../passwords.md) include a built-in password generator that you can use.

### Rotating Passwords

You should avoid changing passwords that you have to remember (such as your password manager's master password) too often unless you have reason to believe it has been compromised, as changing it too often exposes you to the risk of forgetting it.

> When it comes to passwords that you don't have to remember (such as passwords stored inside your password manager), if your [threat model](threat-modeling.md) calls for it, we recommend going through important accounts accounts (especially accounts that don't use multi-factor authentication) and changing their password every couple of months, in case they have been compromised in a data breach that hasn't become public yet. Most password managers allow you to set an expiry date for your password to make this easier to manage.

Reviewer: ```suggestion
When it comes to passwords that you don't have to remember (such as passwords stored inside your password manager), if your [threat model](threat-modeling.md) calls for it, we recommend going through important accounts (especially accounts that don't use multi-factor authentication) and changing their password every couple of months, in case they have been compromised in a data breach that hasn't become public yet. Most password managers allow you to set an expiry date for your password to make this easier to manage.
```

2. As an example, let's say you rolled `2-5-2-6-6`. Look through the [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) for the word that corresponds to `25266`.

3. You will find the word `encrypt`. Write that word down.

4. Repeat this process until your passphrase has as many words as you need, which you should separate with a space.

!!! Warning "Important"

    You should **not** re-roll words until you get a combination of words that appeal to you. The process should be completely random.

If you don't have access to or would prefer to not use real dice, you can use your password manager's built-in password generator, as most of them have the option to generate diceware passphrases in addition to regular passwords.

We recommend using [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) to generate your diceware passphrases, as it offers the exact same security as the original list, while containing words that are easier to memorize. There are also other wordlists in different languages, if you do not want your passphrase to be in English.

> ??? Note "Explanation of entropy and stregth of diceware passphrases"

Reviewer: ```suggestion
??? Note "Explanation of entropy and strength of diceware passphrases"
```

---
title: "Introduction to Passwords"

> icon: 'material-shield-key-outline'

Reviewer 2: ```suggestion
icon: 'material/shield-key-outline'
```

---

> Passwords are an essential part of of our everyday digital lives. We use them to protect our accounts, our devices and our secrets. Despite the fact that passwords are often the only thing between us and someone who wants to access our private information, not a lot of thought is put into them, often leading to using passwords that can be easily guessed or bruteforced and thus jeopardizing your most important data.

Reviewer: ```suggestion
Passwords are an essential part of our everyday digital lives. We use them to protect our accounts, our devices and our secrets. Despite often being the only thing between us and someone who wants to access our private information, not a lot of thought is put into them, often leading to using passwords that can be easily guessed or brute-forced and thus jeopardizing your most important data.
```

A lot of password managers will also notify you when a data breach that is relevant to you has occurred, at which point you should change your password as soon as possible.

## Creating strong passwords

### Diceware

Diceware is a method for creating passphrases which are easy to remember, but hard to guess.

The method traditionally involves using a six-sided die. Every five rolls correspond to a word in a cryptographic list. For example, 43146 in the traditional list corresponds to "munch". You would then continue rolling until your passphrase has as many words as you need, separated by spaces.

The die method ensures that you're generating the passphrase in a completely offline way. If you choose to go that route, you should ideally use unbiased die to ensure that they are as random as possible.

Alternatively, you can use a password generator (your password manager can most likely generate diceware passphrases).

## Storing Passwords

### Password Managers

> The best way to store your passwords is by using a password manager. They allow you to store your passwords in a file or in the cloud with the use of a master password. That way, you will only have to remember one strong password, which lets you access the rest of them.

Reviewer 3: ```suggestion
The best way to store your passwords is by using a password manager. They allow you to store your passwords in a file or in the cloud and protect them with a single master password. That way, you will only have to remember one strong password, which lets you access the rest of them.
```

> There are many good options to choose from, both cloud-based and local. Choose one of our recommended password managers and use it to establish strong passwords across all of your accounts. We recommend securing your password manager with word [diceware](#diceware) passphrase comprised of at least 6 words.

Reviewer 3: ```suggestion
There are many good options to choose from, both cloud-based and local. Choose one of our recommended password managers and use it to establish strong passwords across all of your accounts. We recommend securing your password manager with a word [diceware](#diceware) passphrase comprised of at least 6 words.
```

The die method ensures that you're generating the passphrase in a completely offline way. If you choose to go that route, you should ideally use unbiased die to ensure that they are as random as possible.

Alternatively, you can use a password generator (your password manager can most likely generate diceware passphrases).

## Storing Passwords

### Password Managers

The best way to store your passwords is by using a password manager. They allow you to store your passwords in a file or in the cloud and protect them with a single master password. That way, you will only have to remember one strong password, which lets you access the rest of them.

If you're using TOTP as a [multi-factor authentication](../multi-factor-authentication.md) method for any of your accounts, do not store these tokens in your password manager, as that negates their benefit. You should use a dedicated [TOTP app](../multi-factor-authentication.md/#authenticator-apps) instead.

> There are many good options to choose from, both cloud-based and local. Choose one of our recommended password managers and use it to establish strong passwords across all of your accounts. We recommend securing your password manager with word [diceware](#diceware) passphrase comprised of at least 6 words.

Reviewer 3: ```suggestion
There are many good options to choose from, both cloud-based and local. Choose one of our recommended password managers and use it to establish strong passwords across all of your accounts. We recommend securing your password manager with a [diceware](#diceware) passphrase comprised of at least 6 words.
```

!!! Note

    These instructions assume that you are using [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) to generate the passphrase, which requires five dice rolls per word. Other wordlists may require more or less rolls per word, and may require a different amount of words to achieve the same entropy.

1. Roll a six-sided die five times, noting down the number after each roll.

2. As an example, let's say you rolled `2-5-2-6-6`. Look through the [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) for the word that corresponds to `25266`.

3. You will find the word `encrypt`. Write that word down.

4. Repeat this process until your passphrase has as many words as you need, which you should separate with a space.

!!! Warning "Important"

    You should **not** re-roll words until you get a combination of words that appeal to you. The process should be completely random.

If you don't have access to or would prefer to not use real dice, you can use your password manager's built-in password generator, as most of them have the option to generate diceware passphrases in addition to regular passwords.

> We recommend using [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) to generate your diceware passphrases, as it offers the exact same security as the original list, while containing words that are easier to memorize. There are also other wordlists that in different languages, if you do not want your passphrase to be in English.

Reviewer: ```suggestion
We recommend using [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) to generate your diceware passphrases, as it offers the exact same security as the original list, while containing words that are easier to memorize. There are also other wordlists in different languages, if you do not want your passphrase to be in English.
```

- Your adversary knows that you used the diceware method.
- Your adversary knows the specific wordlist that you used.
- Your adversary knows how many words your passphrase contains.

## Storing Passwords

### Password Managers

The best way to store your passwords is by using a password manager. They allow you to store your passwords in a file or in the cloud and protect them with a single master password. That way, you will only have to remember one strong password, which lets you access the rest of them.

There are many good options to choose from, both cloud-based and local. Choose one of our recommended password managers and use it to establish strong passwords across all of your accounts. We recommend securing your password manager with a [diceware](#diceware) passphrase comprised of at least seven words.

[List of recommended password managers](../passwords.md){ .md-button }

!!! Warning "Don't place your passwords and TOTP tokens inside the same password manager"

>     If you're using TOTP as a [multi-factor authentication](../multi-factor-authentication.md) method for any of your accounts, do not store these tokens, any backup codes for them, or the TOTP secrets themselves in your password manager, as that negates the benefit of multi factor authentication. You should use a dedicated [TOTP app](../multi-factor-authentication.md/#authenticator-apps) instead.

Reviewer: ```suggestion
    If you're using TOTP as a [multi-factor authentication](../multi-factor-authentication.md) method for any of your accounts, do not store these tokens, any backup codes for them, or the TOTP secrets themselves in your password manager, as that negates the benefit of multi-factor authentication. You should use a dedicated [TOTP app](../multi-factor-authentication.md/#authenticator-apps) instead.
```

---
title: "Introduction to Passwords"
icon: 'material/shield-key-outline'
---

> Passwords are an essential part of our everyday digital lives. We use them to protect our accounts, our devices and our secrets. Despite often being the only thing between us and and an adversary who's after our private information, not a lot of thought is put into them which often leads to people using passwords that can be easily guessed or brute-forced.

Reviewer 3: ```suggestion
Passwords are an essential part of our everyday digital lives. We use them to protect our accounts, our devices and our secrets. Despite often being the only thing between us and an adversary who's after our private information, not a lot of thought is put into them which often leads to people using passwords that can be easily guessed or brute-forced.
```

## Best Practices

### Use unique passwords for every service

Imagine this; you sign up for an account with the same e-mail and password on multiple online services. If one of those service providers is malicious, or their service has a data breach that exposes your password in an unencrypted format, all a bad actor would have to do is try that e-mail and password combination across multiple popular services until they get a hit. It doesn't matter how strong that one password is, because they already have it.

> This is called [credential stuffing](https://en.wikipedia.org/wiki/Credential_stuffing) and it is one of the most common ways that your accounts can be compromised by bad actors. To avoid this, make sure that you never re-use the same password twice.

Reviewer 3: ```suggestion
This is called [credential stuffing](https://en.wikipedia.org/wiki/Credential_stuffing) and it is one of the most common ways that your accounts can be compromised by bad actors. To avoid this, make sure that you never re-use your passwords.
```

Reviewer 3: I think the original sentence is redundant.

If you need a password you can memorize, we recommend a [diceware passphrase](#diceware).

### Diceware Passphrases

Diceware is a method for creating passphrases which are easy to remember, but hard to guess.

Diceware passphrases are a great option when you need to memorize or manually input your credentials, such as for your password manager's master password or your device's encryption password.

An example of a diceware passphrase is `viewable fastness reluctant squishy seventeen shown pencil`.

To generate a diceware passphrase using real dice, follow these steps:

!!! Note

    These instructions assume that you are using [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) to generate the passphrase, which requires five dice rolls per word. Other wordlists may require more or less rolls per word, and may require a different amount of words to achieve the same entropy.

> 1. Roll a six-sided dice five times, noting the number after each roll.

Reviewer 3: ```suggestion
1. Roll a six-sided die five times, noting the number after each roll.
```

### Password Managers

The best way to store your passwords is by using a password manager. They allow you to store your passwords in a file or in the cloud and protect them with a single master password. That way, you will only have to remember one strong password, which lets you access the rest of them.

There are many good options to choose from, both cloud-based and local. Choose one of our recommended password managers and use it to establish strong passwords across all of your accounts. We recommend securing your password manager with a [diceware](#diceware) passphrase comprised of at least seven words.

[List of recommended password managers](../passwords.md){ .md-button }

!!! Warning "Don't place your passwords and TOTP tokens inside the same password manager"

    If you're using TOTP as a [multi-factor authentication](../multi-factor-authentication.md) method for any of your accounts, do not store these tokens, any backup codes for them, or the TOTP secrets themselves in your password manager, as that negates the benefit of multi factor authentication. You should use a dedicated [TOTP app](../multi-factor-authentication.md/#authenticator-apps) instead.

### Backups

> You should store an encrypted exported backup of your passwords or your password database itself on multiple storage devices. This can be useful if something happens to your primary device or the service you are using.

Reviewer 3: ```suggestion
You should store an encrypted, exported backup of your passwords or your password database itself on multiple storage devices. This can be useful if something happens to your primary device or the service you are using.
```

### Use unique passwords for every service

Imagine this; you've signed up for an account with your e-mail and the same password on multiple online services. The moment one of those services has a data breach, all a bad actor would have to do is try that e-mail and password combination across multiple popular services until they get a hit. It doesn't matter how strong that one password is, because they already have it.

> This is called [credential stuffing](https://en.wikipedia.org/wiki/Credential_stuffing) and it is one of the most common ways that your accounts can be compromised by bad actors. To avoid this, make sure that you never re-use the same password twice.

Reviewer 4: This can also be done by the owner of the service (https://xkcd.com/792/) which somehow feels more dangerous than a possible (but in fact no so unlikely) data breach.

The dice method ensures that you're generating the passphrase in a completely offline way. Alternatively, you can use a password generator, as most of them have the option to generate diceware passphrases in addition to regular passwords.

Diceware passphrases are great for everything that requires you to memorize or manually input your credentials, such as your password manager's master password or your device's encryption key.

> We recommend using [EFF's large wordlist](https://www.eff.org/files/2016/07/18/eff_large_wordlist.txt) to generate your diceware passphrases, as it offers the exact same security as the original list, while containing words that are easier to memorize. ==Your passphrase should contain at least 6 words separated by spaces to be sufficiently secure.==

Reviewer 4: Outside of the scope of this PR, but what do you think about searching for and recommending non-english lists as well. This will make diceware easier and more accepted by non-native speakers.
